odoo/addons/base/models/res_users.py

```diff
@@ -447,8 +447,9 @@ def onchange_parent_id(self):
 
     def _read(self, fields):
         super(Users, self)._read(fields)
-        canwrite = self.check_access_rights('write', raise_exception=False)
-        if not canwrite and set(USER_PRIVATE_FIELDS).intersection(fields):
+        if set(USER_PRIVATE_FIELDS).intersection(fields):
+            if self.check_access_rights('write', raise_exception=False):
```

Comment:
you forget to inverse the condition no ?

Reply:
Actually my first push was inverted, had a brainfart too.